terraform/app/modules/ecs_service/main.tf

```diff
@@ -0,0 +1,100 @@
+resource "aws_security_group" "this" {
+  name        = "${var.server_type}-service-${var.environment}"
+  description = "Security Group for communication with Background ECS Service"
+  vpc_id      = var.network_params.vpc_id
+  lifecycle {
+    ignore_changes = [description]
+  }
+}
+
+resource "aws_security_group_rule" "egress_all" {
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.this.id
+}
+
+resource "aws_ecs_service" "this" {
+  name                              = "mavis-${var.environment}-${var.server_type}"
+  cluster                           = var.cluster_id
+  task_definition                   = aws_ecs_task_definition.this.arn
+  desired_count                     = var.desired_count
+  launch_type                       = "FARGATE"
+  enable_execute_command            = true
+  health_check_grace_period_seconds = 60
+
+  network_configuration {
+    subnets         = var.network_params.subnets
+    security_groups = [aws_security_group.this.id]
+  }
+  deployment_controller {
+    type = var.deployment_controller
+  }
+  dynamic "deployment_circuit_breaker" {
+    for_each = var.deployment_controller == "ECS" ? [1] : []
+    content {
+      enable   = true
+      rollback = true
+    }
+  }
+  dynamic "load_balancer" {
+    for_each = var.loadbalancer != null ? [1] : []
+    content {
+      target_group_arn = var.loadbalancer.target_group_arn
+      container_name   = "mavis-application"
+      container_port   = var.loadbalancer.container_port
+    }
+  }
+  deployment_minimum_healthy_percent = 100
+  deployment_maximum_percent         = 200
+  lifecycle {
+    ignore_changes = [
+      task_definition,
+      load_balancer,
+      # desired_count TODO: Uncomment once we include autoscaling
+    ]
+    create_before_destroy = true
+  }
+}
+
+resource "aws_ecs_task_definition" "this" {
+  family                   = "${var.server_type}-task-definition-${var.environment}"
+  requires_compatibilities = ["FARGATE"]
+  network_mode             = "awsvpc"
+  cpu                      = 1024
+  memory                   = 2048
+  execution_role_arn       = var.task_config.execution_role_arn
+  task_role_arn            = var.task_config.task_role_arn
+  container_definitions = jsonencode([
+    {
+      name      = "mavis-application"
+      image     = var.task_config.docker_image
```

Comment:
I wonder if it would be nicer to only need to pass in the image tag to the ECS service, I imagine all our services will want to be on the same image URL.

Reply:
Yeah that's fair enough, I think once we have a single repository we can tidy up a few things like this.